Type aliased map gives build error when using delete

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Pop!_OS 22.04 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i7-5600U CPU @ 2.60GHz
CC version: cc (Ubuntu 11.2.0-19ubuntu1) 11.2.0

getwd: /home/hedo/usr/tmp/v
vmodules: /home/hedo/.vmodules
vroot: /home/hedo/usr/tmp/v/v
vexe: /home/hedo/usr/tmp/v/v/v
vexe mtime: 2022-09-02 10:11:47
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.1 4bd49a0

Git version: git version 2.34.1
Git vroot status: 0.3.1-8-g4bd49a01
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 827f7452

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg problem.v`
```v
type Map = map[int]int

fn main() {
	mut m := Map(map[int]int)
	m.delete(0)
}

```

**What did you expect to see?**




**What did you see instead?**
```
==================
/tmp/v_1000/problem.5271690335428903634.tmp.c:12071: warning: implicit declaration of function 'Map_int_int_delete'
tcc: error: undefined symbol 'Map_int_int_delete'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/builtin/map_test.v
+++ b/vlib/builtin/map_test.v
@@ -980,3 +980,16 @@ fn test_map_set_fixed_array_variable() {
 	println(m2)
 	assert '$m2' == "{'A': [1.1, 2.2]}"
 }
+
+type Map = map[int]int
+
+fn test_alias_of_map_delete() {
+	mut m := Map(map[int]int{})
+	m[11] = 111
+	m[22] = 222
+	println(m)
+	m.delete(11)
+	println(m)
+	assert m.len == 1
+	assert m[22] == 222
+}

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -832,8 +832,8 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 		}
 	}
 
-	if left_sym.kind == .map && node.name == 'delete' {
-		left_info := left_sym.info as ast.Map
+	if final_left_sym.kind == .map && node.name == 'delete' {
+		left_info := final_left_sym.info as ast.Map
 		elem_type_str := g.typ(left_info.key_type)
 		g.write('map_delete(')
 		if left_type.has_flag(.shared_f) {

